File: internal/models/recipe.go
package models

import (
	"time"
)

type Recipe struct {
	ID            int       `json:"id"`
	Title         string    `json:"title"`
	Created       time.Time `json:"created"`
	Description   string    `json:"description"`
	UserId        int       `json:"userid"`
	ThumbnailName string    `json:"thumbnail_name"`
	ThumbnailPath string    `json:"thumbnail_path"`
}

File: ui/src/_models/Recipe.ts
export type Recipe = {
  id: number;
  title: string;
  created: string;
  description: string;
  steps: string[];
  version: number;
  user_id: number;
  user_name: string;
};

export type RecipeSimple = {
  id: number;
  title: string;
};
